### ISSUE TITLE
lerna bootstrap hangs aftet first run when hoist is on

### ISSUE THREAD

**Issue by noherczeg** [lerna/lerna#582]
Modules and deps:
```
- A
- B
- C
- D
    - A (dep)
    - B (dep)
    - C (dep)
    - F (dep)
- E
    - B (peer dep)
    - C (peer dep)
    - F (peer dep)
- F
    - B (peer dep)
```

lerna.json:
```
{
  "lerna": "2.0.0-beta.36",
  "packages": [
    "modules/@whatevs/*"
  ],
  "hoist": true,
  "version": "x.y.z"
}
```

When I first run `lerna bootstrap` it runs fine, but every subsequent runs just hang at "Prune hoisted". It also kills my PC basically. Some heavy I/O and/or CPU ops might be running.

I can not produce any logs since it just hangs, and I need to kill the process :(

@gigabo Did you encounter anything like this?

**Comment by noherczeg** [comment]
Removing `node_modules` in each sub-module didn't help either.

**Comment by noherczeg** [comment]
@gigabo Could you ping me if the fix got released? :)

**Comment by ivanov** [comment]
I'm seeing hangs with `--hoist` flag as well. :\

**Comment by noherczeg** [comment]
@ivanov yesterday @gigabo told me that it is fixed in master. We just need to wait for it to arrive in the next release.

**Comment by kmalakoff** [comment]
I'm also encountering a hang when had lerna with hoist in postinstall. Running it manually after npm install avoided the hang.

**Comment by gigabo** [comment]
Sorry for the trouble here.  This should be resolved in [`lerna@2.0.0-beta.37`](https://github.com/lerna/lerna/releases/tag/v2.0.0-beta.37).

**Comment by noherczeg** [comment]
Yes it is, thanks @gigabo :)
